client/src/main/java/org/schemarepo/client/Avro1124RESTRepositoryClient.java

```diff
@@ -93,7 +93,7 @@ public Subject lookup(String subject) {
   public Iterable<Subject> subjects() {
     ArrayList<Subject> subjectList = new ArrayList<Subject>();
     try {
-      String subjects = webResource.get(String.class);
+      String subjects = webResource.accept(MediaType.TEXT_PLAIN_TYPE).get(String.class);
```

Comment:
It is hard to gauge the use of AVRO-1124 since it is a back-end component. Here are some indicators, however (as of 2014-01-14) :

* The [Apache JIRA ticket](https://issues.apache.org/jira/browse/AVRO-1124) has:
    * 30 votes, which ranks it:
        * [20th of all time among all Apache projects](https://issues.apache.org/jira/issues/?jql=%20order%20by%20votes%20desc)
        * [4th in currently open tickets among all Apache projects](https://issues.apache.org/jira/issues/?jql=status%20%3D%20OPEN%20order%20by%20votes%20desc) 
        * [1st of all time in AVRO](https://issues.apache.org/jira/issues/?jql=project%20%3D%20AVRO%20order%20by%20votes%20desc)
    * 61 watchers, which ranks it:
        * [36th of all time among all Apache projects](https://issues.apache.org/jira/issues/?jql=order%20by%20watchers%20desc).
        * [11th in currently open tickets among all Apache projects](https://issues.apache.org/jira/issues/?jql=status%20%3D%20open%20order%20by%20watchers%20desc)
        * [1st of all time in AVRO](https://issues.apache.org/jira/issues/?jql=project%20%3D%20AVRO%20order%20by%20watchers%20desc)
    * A flurry of comments, spanning 2.5 years.
* The [Camus](https://github.com/linkedin/camus) project uses it as a component.
    * Camus is (probably) the most popular way to move data from Kafka to Hadoop, having 94 watchers, 377 stars, 237 forks and 26 unique contributors on GitHub.
    * The term "1124" has been [mentioned in 18 different threads on the Camus mailing list](https://groups.google.com/forum/#!searchin/camus_etl/1124). Most of those threads have several people participating and hundreds of views each (according to Google Groups anyway, which presumably includes only web views, not opened emails, since that is never properly traceable).

The schema repo is meant to make evolution of schemas easier. It is all about backwards and forwards compatibility. Thus, any change that requires a "big bang" style of integration (where both clients and server need to be upgraded simultaneously) will severely impact its usability and growth.

Offering different results based on the UA seems brittle, and goes against the value proposition of REST (being language-agnostic)... I don't see much of a problem with offering an HTML view of the repo on a different URL root, however... Having it all under one root is a nice to have, but correctness and compatibility are more important, IMHO.